### ISSUE TITLE
Clarify use for passwords

### ISSUE THREAD

**Issue by lightonflux** [minetest/minetest#1043]
It is very confusing to new users when a server requires a password, just look in the forum or chats. That is because most people expect that the password is for accessing the server, not to "secure" their username.

I think it would be helpful if a mouse-over or little help-icon would explain for what the password is and what it secures.

**Comment by Wuzzy2** [comment]
Would probably be fixed by #2438. Your opinion?

**Comment by lightonflux** [comment]
Looka good but could be more clear:

>       tooltips.te_pwd = fgettext([[Password for the server, it is accociated with the name. Leave this field empty if you don't want to use a password. WARNING: Setting the password for the first time on a server will automatically create an account on the server and you need to use the same password afterwards.]])


Instead something like:

> **Password for your account on the server, it prevents other people from using your account.** Leave this field empty if you don't want to use a password. WARNING: Setting the password for the first time on a server will automatically create an account on the server and you need to use the same password afterwards.

**Comment by Wuzzy2** [comment]
Jeez, this is even longer than the current tooltip. Yes, this one is more clear, but many developers think the tooltips are already too long. I guess increasing the tooltip length would lower the chances of merging even further. :-(

**Comment by lightonflux** [comment]
Is the warning really necessary? Because if somebody knows what an account and password is from the web then it really isn't necessary.

**Comment by lightonflux** [comment]
How about this? :D
> Password for your account on the server, it prevents other people from using your account. If you don't like protection leave it empty.

**Comment by lightonflux** [comment]
Tried to lookup the tooltip in the menu (compiled latest git) and i don't get tool tips. 

Are tool tips being reworked?

**Comment by paramat** [comment]
This type of information is only needed once, after that it becomes unnecessary clutter for players. So i feel it should be elsewhere in a text file somewhere, not a tooltip or pop-up.

**Comment by Wuzzy2** [comment]
I no longer think this issue can be really solved just by a tooltip alone. There a bigger underlying issues here. Maybe the whole UI around this password thing must be rethought. It is bad for several reasons:

* What lightonflux said
* On first-time login, it *silently* creates account and protects it with password—It can cost a newbie several “burned” accounts until finally realizing that, since many servers (sadly) force users to enter passwords. It also took me a while until I finally “got” how it works. It's frustrating when you lose a good user name because of that.
* On the other hand, there is no obvious way to know whether an account creation *failed* because the server is down. Maybe the packet arrived on the server but then the connection broke down at this exact moment
* No visible difference beween account *creation* and a *normal login*. It all happens “magically” and silently without the user being noticed at all
* No requirement to enter a password twice on account creation. If you screwed up your password in the first login (aka the silent magic account creation), you have burned just another account
* No way to know whether you are forced to enter a password or not until you try. Very annoying if you just want to quickly peek into a server (maybe this forced password thing should be dropped, but thats a different issue)

If you compare Minetest's login and account system to the account system to basically *every website* in the web, you will notice:
* You will rarely see a website where the login dialog is the same as the register dialog. Login and registration are almost always two different things (from the user perspective)
* Almost every registration dialog has two password fieds to avoid screwing up your account very early
* Also, almost every website notifies the user right after account creation

All this is pretty basic stuff. There is zero reason for Minetest not to follow this de facto standard.

Well, so more thoughtwork needs to go into the UI, but I am myself not in the right mood for this either (sorry :-( ).

**Comment by Ferk** [comment]
Maybe the biggest obstacle for this is not really the UI, but the network protocol.

I assume the client doesn't really have a way to know in advance if a username is already exiting in the server or needs creation (and for security concerns, it shouldn't). And by the looks of it there doesn't seem to be a way to explicitly tell the server you want to login only if the username exists (without creating it).

**Comment by rubenwardy** [comment]
Duplicate of #258
